### ISSUE TITLE
Fix `selector-no-qualifying-type` false negatives for `:is/where()`

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#8927]
### What minimal example or steps are needed to reproduce the bug?

```css
#foo:is(.bar a) {}
#foo:where(.bar a) {}
:is(a, .foo):is(b, .bar) {}
```

The first `:is()` one is also what standard nesting desugars the following into:

```css
.bar a { &#foo {} }
```

### What minimal configuration is needed to reproduce the bug?

```mjs
export default {
	rules: {
		"selector-no-qualifying-type": true
	}
};

```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIDEAzCF4EsDOAKAdAEYCGATgATECU5wAvgDoB2q68A7gBYykwEkVqtRkyx5iAGnL40EKmNyEpRMjXogJIFJgA2MAHLEAtnEQwAHsYAOe-GGzYN4CE20BzBCAtWIpAC7ksCjEAK46AcDMDH6kYTDY8LRRftEMINgwemB+vgC0TBC5AI4hxDqYKACemExuuX6VVjBpiTEhzUzRInQA3MxOkK6YbgBivkbEfp7YDXrlTH52Lu74RgBWjpqwVo6IkUzk5GkzlXM1KQhHIDqT8RcSzIfHs5nnuYPuuTPETFBkUC0rjc-Hc0sw6CA6EA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.26.1",
  "stylelint-config-standard": "39.0.1"
}
```

### What did you expect to happen?

The `a` and `b`s in the selectors above to be flagged as qualifying.

### What actually happened?

No problems reported.

### Do you have a proposal to fix the bug?

Check the rightmost of the applicable function pseudos. See https://github.com/stylelint/stylelint/pull/8913#discussion_r2618599821

Branch off https://github.com/stylelint/stylelint/pull/8913 or wait for it to be merged in `v17`.
